langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiClient.java

```diff
@@ -0,0 +1,206 @@
+            // Log raw HTTP requests
+            if (logRequests) {
+                okHttpClientBuilder.addInterceptor(new MistralAiRequestLoggingInterceptor());
+            }
+
+            // Log raw HTTP responses
+            if (logResponses) {
+                okHttpClientBuilder.addInterceptor(new MistralAiResponseLoggingInterceptor());
+            }
+        }
+
+        this.okHttpClient = okHttpClientBuilder.build();
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(baseUrl)
+                .client(okHttpClient)
+                .addConverterFactory(GsonConverterFactory.create(GSON))
+                .build();
+
+        mistralAiApi = retrofit.create(MistralAiApi.class);
+    }
+
+    public MistralChatCompletionResponse chatCompletion(MistralChatCompletionRequest request) {
+        try {
+            retrofit2.Response<MistralChatCompletionResponse> retrofitResponse
+                    = mistralAiApi.chatCompletion(request).execute();
+            if (retrofitResponse.isSuccessful()) {
+                LOGGER.debug("ChatCompletionResponseBody: {}", retrofitResponse.body());
+                return retrofitResponse.body();
+            } else {
+                throw toException(retrofitResponse);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void streamingChatCompletion(MistralChatCompletionRequest request, StreamingResponseHandler<AiMessage> handler) {
+        EventSourceListener eventSourceListener = new EventSourceListener() {
+            final StringBuffer contentBuilder = new StringBuffer();
+            TokenUsage tokenUsage;
+            FinishReason finishReason;
+
+            @Override
+            public void onOpen(EventSource eventSource, okhttp3.Response response) {
+                LOGGER.debug("onOpen()");
+            }
+
+            @Override
+            public void onEvent(EventSource eventSource, String id, String type, String data) {
+
+                LOGGER.debug("onEvent() {}", data);
+                if ("[DONE]".equals(data)) {
+                    Response<AiMessage> response = Response.from(
+                            AiMessage.from(contentBuilder.toString()),
+                            tokenUsage,
+                            finishReason
+                    );
+                    handler.onComplete(response);
+                } else {
+                    try {
+                        MistralChatCompletionResponse chatCompletionResponse = GSON.fromJson(data, MistralChatCompletionResponse.class);
+                        MistralChatCompletionChoice choice = chatCompletionResponse.getChoices().get(0);
+                        String chunk = choice.getDelta().getContent();
+                        contentBuilder.append(chunk);
+                        handler.onNext(chunk);
+
+                        MistralUsageInfo usageInfo = choice.getUsage();
```

Comment:
```suggestion
                        MistralUsageInfo usageInfo = chatCompletionResponse.getUsage();
```
I made a mistake in this code. Right now `MistralAiStreamingChatModelIT` fails because of this. Changing `choice` to `chatCompletionResponse` should fix it.